PORTAL/jobs/_pyperformance.py

```diff
@@ -688,18 +790,18 @@ def __eq__(self, other):
             return False
         return True
 
-    def _as_hashable(self):
+    def _as_hashable(self) -> Tuple[Any, Tuple]:
         return (
             self._source,
             tuple(sorted(self._byname.items())) if self._byname else (),
         )
 
     @property
-    def source(self):
+    def source(self) -> str:
         return self._source
 
     @property
-    def byname(self):
+    def byname(self) -> Dict[str, _utils.ElapsedTimeWithUnits]:
```

Comment:
Again, the type depends on the return type of `cls._parse_value()`, which can be different for subclasses.